Classes/Reporter/BaseReporter.php

```diff
@@ -0,0 +1,173 @@
+use Psr\Http\Message\ResponseInterface;
+use GuzzleHttp\Exception\RequestException;
+
+abstract class BaseReporter extends CrawlObserver
+{
+    /**
+     * @Flow\Inject
+     * @var ResultItemStorage
+     */
+    protected $resultItemStorage;
+
+    /**
+     * @Flow\Inject
+     * @var ConsoleOutput
+     */
+    protected $output;
+
+    /**
+     * @Flow\InjectConfiguration(path="excludeStatusCodes")
+     */
+    protected array $excludeStatusCodes = [];
+
+    protected array $resultItemsGroupedByStatusCode = [];
+
+    public function getResultItemsGroupedByStatusCode(): array
+    {
+        return $this->resultItemsGroupedByStatusCode;
+    }
+
+    public function __construct()
+    {
+    }
+
+    /**
+     * Outputs specified text to the console window and appends a line break
+     *
+     * @see output()
+     * @see outputLines()
+     */
+    protected function outputLine(string $text = '', array $arguments = []): void
+    {
+        $this->output->outputLine($text, $arguments);
+    }
+
+    /**
+     * Called when the crawler has crawled the given url successfully.
+     *
+     * @return int
+     */
+    public function crawled(
+        UriInterface $url,
+        ResponseInterface $response,
+        ?UriInterface $foundOnUrl = null
+    ) {
+        $statusCode = $response->getStatusCode();
+        if (!$this->isExcludedStatusCode($statusCode)) {
+            $this->addCrawlingResultToStore($url, $foundOnUrl, $statusCode);
+        }
+
+        return $statusCode;
+    }
+
+    /**
+     * Called when the crawler had a problem crawling the given url.
+     */
+    public function crawlFailed(
+        UriInterface $url,
+        RequestException $requestException,
+        ?UriInterface $foundOnUrl = null
+    ): int {
+        $statusCode = (int)$requestException->getCode();
+        if (!$this->isExcludedStatusCode($statusCode)) {
+            $this->addCrawlingResultToStore($url, $foundOnUrl, $statusCode);
+        }
+
+        return $statusCode;
+    }
+
+    /**
+     * We collect the crawling results in the class variable urlsGroupedByStatusCode.
+     * We store the crawled url, the status code for this url and if an origin url exists also the location where
+     * we got the crawling url from.
+     */
+    protected function addCrawlingResultToStore(
+        UriInterface $crawlingUrl,
+        UriInterface $originUrl = null,
+        int $statusCode = 200
+    ): void {
+        $cliMessage = "Checked {$crawlingUrl} from {$originUrl} with status {$statusCode}";
+        if ($originUrl === null) {
+            $cliMessage = "Checked {$crawlingUrl} with status {$statusCode}";
+        }
+
+        $this->outputLine($cliMessage);
+
+        if ($statusCode === 200) {
+            return;
+        }
+
+        $parts = parse_url((string)$originUrl);
+
+        if ($parts === false) {
+            return;
+        }
+
+        $linkCheckItem = new ResultItem();
+        $linkCheckItem->setDomain($parts['host']);
```

Comment:
Hosts ist nicht immer gesetzt, hier meine Fehlermeldung:


Exception #1 in line 121 of /var/www/html/Packages/Application/CodeQ.LinkChecker/Classes/Reporter/BaseReporter.php: Notice: Undefined index: host in /var/www/html/Packages/Application/CodeQ.LinkChecker/Classes/Reporter/BaseReporter.php line 121

34 Neos\Flow\Error\ErrorHandler::handleError(8, "Undefined index: host", "/var/www/html/Packages/Application/CodeQ.LinkChecker/Classes/Reporter/BaseReporter.php", 121, array|6|)
33 CodeQ\LinkChecker\Reporter\BaseReporter::addCrawlingResultToStore(GuzzleHttp\Psr7\Uri, NULL, 302)
32 CodeQ\LinkChecker\Reporter\BaseReporter::crawled(GuzzleHttp\Psr7\Uri, GuzzleHttp\Psr7\Response, NULL)
31 Spatie\Crawler\CrawlObserverCollection::crawled(Spatie\Crawler\CrawlUrl, GuzzleHttp\Psr7\Response)
30 Spatie\Crawler\Handlers\CrawlRequestFulfilled::handleCrawled(GuzzleHttp\Psr7\Response, Spatie\Crawler\CrawlUrl)
29 Spatie\Crawler\Handlers\CrawlRequestFulfilled::__invoke(GuzzleHttp\Psr7\Response, "https://neos.io/", GuzzleHttp\Promise\Promise)
28 call_user_func(Spatie\Crawler\Handlers\CrawlRequestFulfilled, GuzzleHttp\Psr7\Response, "https://neos.io/", GuzzleHttp\Promise\Promise)
27 GuzzleHttp\Promise\EachPromise::GuzzleHttp\Promise\{closure}(GuzzleHttp\Psr7\Response)
26 GuzzleHttp\Promise\Promise::callHandler(1, GuzzleHttp\Psr7\Response, NULL)
25 GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
24 GuzzleHttp\Promise\TaskQueue::run()
23 GuzzleHttp\Handler\CurlMultiHandler::tick()
22 GuzzleHttp\Handler\CurlMultiHandler::execute(true)
21 GuzzleHttp\Promise\Promise::invokeWaitFn()
20 GuzzleHttp\Promise\Promise::waitIfPending()
19 GuzzleHttp\Promise\Promise::invokeWaitList()
18 GuzzleHttp\Promise\Promise::waitIfPending()
17 GuzzleHttp\Promise\Promise::wait()
16 GuzzleHttp\Promise\EachPromise::GuzzleHttp\Promise\{closure}(true)
15 GuzzleHttp\Promise\Promise::invokeWaitFn()
14 GuzzleHttp\Promise\Promise::waitIfPending()
13 GuzzleHttp\Promise\Promise::wait()
12 Spatie\Crawler\Crawler::startCrawlingQueue()
11 Spatie\Crawler\Crawler::startCrawling(GuzzleHttp\Psr7\Uri)
10 CodeQ\LinkChecker\Command\CheckLinksCommandController_Original::crawlCommand(10)
9 Neos\Flow\Cli\CommandController_Original::callCommandMethod()
8 Neos\Flow\Cli\CommandController_Original::processRequest(Neos\Flow\Cli\Request, Neos\Flow\Cli\Response)
7 Neos\Flow\Cli\Dispatcher_Original::dispatch(Neos\Flow\Cli\Request, Neos\Flow\Cli\Response)
6 Neos\Flow\Cli\CommandRequestHandler::Neos\Flow\Cli\{closure}()
5 Closure::__invoke()
4 Neos\Flow\Security\Context_Original::withoutAuthorizationChecks(Closure)
3 Neos\Flow\Cli\CommandRequestHandler::handleRequest()
2 Neos\Flow\Core\Bootstrap::run()
1 require("/var/www/html/Packages/Framework/Neos.Flow/Scripts/flow.php")

Reply:
How can I reproduce this? It never happened to me